crates/release_plz/src/args/mod.rs

```diff
@@ -71,22 +76,66 @@ fn parse_config(config_path: Option<&Path>) -> anyhow::Result<Config> {
             },
         }
     } else {
-        match first_file_contents([
+        let file_contents = first_file_contents([
             Path::new("release-plz.toml"),
             Path::new(".release-plz.toml"),
-        ])? {
-            Some((config, path)) => (config, path),
-            None => {
-                info!("release-plz config file not found, using default configuration");
-                return Ok(Config::default());
-            }
+        ])?;
+        if let Some(project_manifest) = project_manifest
+            .or(Some(Path::new(CARGO_TOML)))
+            .filter(|v| v.exists())
+        {
+            return match parse_config_from_metadata(cargo_metadata)? {
+                Some(config) => {
+                    if file_contents.is_some() {
+                        return Err(anyhow::anyhow!(
+                            "configuration is both specified as file and metadata"
+                        ));
+                    }
+                    info!(
+                        "using configuration from metadata in {:?}",
+                        project_manifest
+                    );
+                    Ok(config)
+                }
+                None => Ok(Config::default()),
+            };
+        }
+        if let Some((config, path)) = file_contents {
+            (config, path)
+        } else {
+            info!("release-plz config file not found, using default configuration");
+            return Ok(Config::default());
         }
     };
 
     info!("using release-plz config file {}", path.display());
     toml::from_str(&config).with_context(|| format!("invalid config file {config_path:?}"))
 }
 
+fn parse_config_from_metadata(metadata: &Metadata) -> anyhow::Result<Option<Config>> {
+    // check both workspace and package metadata
+    for metadata in [
+        Some(metadata.clone().workspace_metadata),
+        metadata.packages.first().map(|v| v.metadata.clone()),
```

Comment:
why do you only consider the first package?
Also, `metadata.packages` also contains your dependencies. We should only consider workspace members.
You can find workspace members with [this](https://github.com/orhun/release-plz/blob/471de295c9a01684bde6bd647da6f7290a0fc4ee/crates/cargo_utils/src/workspace_members.rs#L14) function.